db/model/subject.js

```diff
@@ -516,9 +523,9 @@ module.exports = function subject(seq, dataTypes) {
          * unpublished too. If any of the children are published, throw a
          * validation error
          */
-        if (inst.getDataValue('isPublished') === false) {
-          return new seq.Promise((resolve, reject) => {
-            inst.getChildren()
+        function checkPublished() {
```

Comment:
The entire beforeUpdate was refactored. It is best to view the entire file to see the changes.
The un-publish bug was caused by checking for isPublished===false returning the instance without checking for name or parent updates. The refactored version does:

checkIsPublished() <-- throws error if unpublishing a parent, else returns the instance
.then((inst) => updateParent(inst))
.then((inst) => updateName(inst))

This way all three checks are completed in the expected order.